frontend_app_file/src/container/PopupCreateFile.jsx

```diff
@@ -52,10 +52,17 @@ class PopupCreateFile extends React.Component {
   }
 
   componentWillUnmount () {
-    // console.log('%c<File> will Unmount', `color: ${this.state.config.hexcolor}`)
     document.removeEventListener(CUSTOM_EVENT.APP_CUSTOM_EVENT_LISTENER, this.customEventReducer)
   }
 
+  componentDidUpdate (prevProps, prevState, snapshot) {
+    const { state } = this
+
+    if (state.uploadedFileList.length === state.uploadFileList.length && state.uploadStarted) {
```

Comment:
I don't think this is the best way of doing.
I would think you have, somewhere, a list of file to upload on which you'll call an upload function. This upload function should return a promise and you should call `handleUploadEnd` once every promises have finished. That ways you'll be able to handle errors more easily depending on if the promise resolved or rejected